chain-impl-mockchain/src/testing/data/leader.rs

```diff
@@ -0,0 +1,51 @@
+use crate::leadership::bft::LeaderId;
+use chain_crypto::{Ed25519Extended, SecretKey, KeyPair};
+use std::fmt::{self, Debug};
+use quickcheck::{Arbitrary,Gen};
+
+
+#[derive(Clone)]
+pub struct LeaderPair {
+    leader_id: LeaderId,
+    leader_key: SecretKey<Ed25519Extended>,
+}
+
+impl PartialEq<LeaderPair> for LeaderPair {
+    fn eq(&self, other: &LeaderPair) -> bool {
+        self.id() == other.id()
```

Comment:
This looks very dodgy as a `PartialEq` impl.
There is nothing that prevents construction of pairs with the same ID but different keys, so I'd rather remove this and compare the IDs directly where needed.